v3/src/components/case-table/case-table.tsx

```diff
@@ -63,12 +66,32 @@ export const CaseTable = observer(function CaseTable({ setNodeRef }: IProps) {
 
   const handleNewCollectionDrop = useCallback((dataSet: IDataSet, attrId: string, beforeCollectionId: string) => {
     if (dataSet.attrFromID(attrId)) {
+      let collection: ICollectionModel | undefined
+
+      // Determine if the old collection will become empty and therefore get removed
+      const oldCollectionId = dataSet.getCollectionForAttribute(attrId)?.id
+      let removedOldCollection = false
+      if (oldCollectionId) {
+        if (oldCollectionId === dataSet.ungrouped.id) {
+          if (dataSet.ungroupedAttributes.length <= 1) removedOldCollection = true
+        } else {
+          const oldCollectionLength = dataSet.getGroupedCollection(oldCollectionId)?.attributes.length
+          if (oldCollectionLength && oldCollectionLength <= 1) removedOldCollection = true
+        }
+      }
```

Comment:
Instead of duplicating the logic from the `DataSet`, how about calling `getCollection(oldCollectionId)` after calling `moveAttributeToNewCollection` and if it's not there you know it's been removed?

Reply:
I definitely like this idea, but it isn't working for the ungrouped collection. I added a TODO about revisiting this after the collection code gets shuffled.